compute_endpoint/globus_compute_endpoint/engines/base.py

```diff
@@ -117,6 +119,55 @@ def _status_report(self, shutdown_event: threading.Event, heartbeat_period: floa
             packed = messagepack.pack(status_report)
             self.results_passthrough.put({"message": packed})
 
+    def _handle_task_exception(
+        self,
+        task_id: str,
+        execution_begin: TaskTransition,
+        exception: BaseException,
+    ) -> bytes:
+        """Repackage task exception to messagepack'ed bytes
+        Parameters
+        ----------
+        task_id: str
+        execution_begin: TaskTransition
+        exception: Exception object from the task failure
+
+        Returns
+        -------
+        bytes
+        """
+        code, user_message = get_result_error_details(exception)
+        error_details = {"code": code, "user_message": user_message}
+        execution_end = TaskTransition(
+            timestamp=time.time_ns(),
+            actor=ActorName.INTERCHANGE,
+            state=TaskState.EXEC_END,
+        )
+        exception_string = ""
+        for index, prev_exc in enumerate(
+            self._retry_table[task_id]["exception_history"]
+        ):
+            templated_history = f"""
+++++++++++++++++++++ Traceback from attempt:{index} ++++++++++++++++++++
+{get_error_string(exc=prev_exc)}
+------------------------------------------------------------------
+            """
+            exception_string += templated_history
+
+        exception_string += get_error_string(exc=exception)
+
+        result_message = dict(
+            task_id=task_id,
+            data=exception_string,
+            exception=exception,
```

Comment:
`exception=` needs to be a string.  Nice cleanup to only do the work once, however!

Reply:
TBH, I looked everywhere and I can't find any place where that field is defined. I don't see it being used in the SDK either. I've updated it to `exception=exception_string` which is what `execute_task` does. At this point, I suspect the right choice is to remove the exception field.

Reply:
Ah; good research.  I was merely looking at the call-site of what came before and matching data-types.  If it's entirely unused, then certainly we could remove it.  Do you want to do that in this PR?  Or make that a separate PR so as to keep this one more focused?

Reply:
I'll do this in a separate PR